src/machine/machine_rp2040_adc.go

```diff
@@ -20,43 +39,112 @@ func InitADC() {
 	waitForReady()
 }
 
-// Configure configures a ADC pin to be able to be used to read data.
+// Below methods are for standard tinygo, ADC{} pin based access to the ADC peripheral.
+
+// The Configure method sets the ADC pin to analog input mode.
+// Note the ADCConfig parameter is ignored.
+// Does nothing if the GPIO is not an ADC muxed pin.
 func (a ADC) Configure(config ADCConfig) {
+	if c, ok := a.GetADCChannel(); ok {
+		c.Configure(ADCChConfig{})
+	}
+}
+
+// The Get method returns a one-shot ADC sample reading.
+func (a ADC) Get() uint16 {
+	if c, ok := a.GetADCChannel(); ok {
+		return c.GetOnce()
+	}
+	// Not an ADC pin!
+	return 0
+}
+
+// The GetADCChannel method returns the channel associated with the ADC pin.
+// Returns !ok if the pin is not an ADC pin
+func (a ADC) GetADCChannel() (c ADCChannel, ok bool) {
```

Comment:
Probably cleaner to return a package level error with precise information of what went wrong instead of `ok bool`.

`var errBadPinValue = errors.New("no associated pin with value")`

Reply:
agree with @soypat return better to return `error`